### PR TITLE
Add oauth2_proxy

### PR DESCRIPTION
**Pull request by gouthamve**
Signed-off-by: Goutham Veeramachaneni <cs14btech11014@iith.ac.in>

### CODE REVIEW

File: prometheus-ksonnet/lib/nginx.libsonnet [hunk 1]
```diff
@@ -79,4 +80,22 @@
...
+
+          redirect_url: $._config.oauth_redirect_url,
+          upstream: 'http://nginx.%(namespace)s.svc.%(cluster_dns_suffix)s/' % $._config,
+          client_id: $._config.oauth_client_id,
+          client_secret: $._config.oauth_client_secret,
+          cookie_secret: $._config.oauth_cookie_secret,
+          email_domain: $._config.oauth_email_domain,
+        },
+      }
+    ),
```
[the remark this answers is not in the document]
Comment (gouthamve) [reply prometheus-ksonnet/lib/nginx.libsonnet]: Different subdir as in `kausal/public/ksonnet-mixins/oauth2_proxy` or `kausal/public/prometheus-ksonnet/lib/oauth2_proxy`?

**Comment by tomwilkie** [comment]
kausal/public/oauth-proxy?

And I’ll move this repo to grafana.

**Comment by gouthamve** [comment]
This is ready for another review.

**Comment by tomwilkie** [comment]
@gouthamve I had a fiddle with this myself; see the changes.  Let me know if this looks okay.

**Comment by gouthamve** [comment]
LGTM

**Pull request merged**